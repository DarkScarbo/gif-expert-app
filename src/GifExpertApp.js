import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
import PropTypes from 'prop-types';

export const GifExpertApp = ( {defaultCategories = ["Rupaul", "Miss Vanjie"]} ) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            < AddCategory setCategories={setCategories}/>
            <hr/>

            <ol>
                {categories.map(category => (
                <GifGrid key= {category} category= {category}/>
                ))
            }
            </ol>
        </>
    )
}

GifExpertApp.propTypes = {
    categories: PropTypes.array
}