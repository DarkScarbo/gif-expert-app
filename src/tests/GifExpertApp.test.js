import React from 'react'
import '@testing-library/jest-dom';
import { GifExpertApp } from "../GifExpertApp";
const { shallow } = require("enzyme")

describe('Tests in <GifExpertApp />', () => {
    
    const categories = ["Rupaul","Drag Queen "]
    const wrapper = shallow( <GifExpertApp categories= { categories }/>);

    test('should show the component correctly', () => {  
        expect(wrapper).toMatchSnapshot();
    })

    test('should show a list of categories', () => {

        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
})
