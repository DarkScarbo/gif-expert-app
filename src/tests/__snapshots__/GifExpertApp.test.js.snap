// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp /> should show the component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Rupaul"
      key="Rupaul"
    />
    <GifGrid
      category="Miss Vanjie"
      key="Miss Vanjie"
    />
  </ol>
</Fragment>
`;
